src/adapter/baseAdapter.ts

```diff
@@ -1,5 +1,7 @@
 import Gio from 'gi://Gio';
```

Comment:
Maybe it's a naive question, but can we just pass the extension context once to the settings module in an `initialize` function and store it in a variable there? Passing the extension to every settings object looks a bit cumbersome, IMO.

Reply:
I thought two steps ahead around three corners and was under the impression that modules separate them across the importing files.
Much cleaner now!